data/Store.js

```diff
@@ -336,7 +336,7 @@ export class Store extends HoistBase {
     /**
      * Re-runs the Filter on the current data. Applications only need to call this method if
      * the state underlying the filter, other than the record data itself, has changed. Store will
-     * re-filter automatically whenever Record data is updated or modified.
+     * re-filter automatically whenever StoreRecord data is updated or modified.
```

Comment:
I guess we could also move some of these references from 'Record' to 'record'.  Might improve readability in some cases --  'Record'  as a class-name is really what we need to avoid.   

Not a huge deal -- if any of the usages in docs seems especially clunky, something to consider.